File: 3de year project/Woodrock-Animal-Rescue-master/BLL/MedicalProcedure.php
<?php
$servername = "localhost";
$dbname = "WoodRock";

$conn = new mysqli($servername, $username, $password, $dbname);
if ($conn->connect_error) 
{
    die("Connection failed: " . $conn->connect_error);
}

class MedicalProcedure
{
    public $procedureID;
    public $description;
    public $duration;
    public $cost;

    function _constructor($procedureID, $description, $duration, $cost)
    {
        $this ->procedure = $procedureID;
        $this ->description = $description;
        $this ->duration = $duration;
        $this ->cost = $cost;
    }

    function _AddProcedure($procID, $descrip, $dura, $cost)
    {
        $sql = ("INSERT INTO MedicalProcedure (ProcedureID, Description, Duration, Cost)  VALUES ('{0}','{1}','{2}','{3}') ", 
        $procID, $descrip, $dura, $cost);

        if ($conn->query($sql) === TRUE) {
            echo "New record created successfully";
        } else {
            echo "Error: " . $sql . "<br>" . $conn->error;
        }
    }

    function _EditProcedure($procID, $descrip, $dura, $cost)
    {
        $sql = ("UPDATE MedicalProcedure SET Description = '{1}',Duration = '{2}',Cost = '{3}' WHERE ProcedureID = '{0}'", $procID, $descrip, $dura, $cost);

        if ($conn->query($sql) === TRUE) {
            echo "Record updated successfully";
        } else {
            echo "Error updating record: " . $conn->error;
        }
    }

    function _RemoveProcedure($procID)
    {
        $sql = ("DELETE FROM MedicalProcedure WHERE ProcedureID = {0}", $procID);

        if (mysqli_query($conn, $sql)) {
            echo "Record deleted successfully";
        } else {
            echo "Error deleting record: " . mysqli_error($conn);
        }
    }
    
    function _ScheduleProcedure($empID, $animalId, $dateOfPro)
    {
        $sql = ("INSERT INTO MedicalProcedures (EmployeeID, AnimalID, Date) VALUES ('{0}','{1}','{2}')", $empID, $animalId, $dateOfPro);

        if ($conn->query($sql) === TRUE) {
            echo "New record created successfully";
        } else {
            echo "Error: " . $sql . "<br>" . $conn->error;
        }    
    }
}

File: 3de year project/Woodrock-Animal-Rescue-master/BLL/MedicalProcedure.cs
using System;
using System.Collections.Generic;
using System.Text;
using Woodrock_Animal_Rescue.DAL;

namespace Woodrock_Animal_Rescue.BLL
{

    class MedicalProcedure
    {
        #region Properties

        private int procedureID;
        private string description;
        private int duration;
        private double cost;

        public global::System.Int32 ProcedureID { get => procedureID; set => procedureID = value; }
        public global::System.String Description { get => description; set => description = value; }
        public global::System.Int32 Duration { get => duration; set => duration = value; }
        public global::System.Single Cost { get => cost; set => cost = value; }
        #endregion

        #region Constructor
        public MedicalProcedure(global::System.Int32 procedureID, global::System.String description, global::System.Int32 duration, global::System.Single cost)
        {
            this.procedureID = procedureID;
            this.description = description;
            this.duration = duration;
            this.cost = cost;
        }
        public MedicalProcedure()
        {

        }
        #endregion

        #region Methods
        public override ToString()
        {
            return string.format("Procedure ID : {0}, Description: {1}, Duration: {2}, Cost: {3}", procedureID,description, duration, cost);
        }

        public void AddProcedure(int procID, string descrip, int dura, double cost)
        {
            string query= string.Format("INSERT INTO MedicalProcedure (ProcedureID, Description, Duration, Cost)  VALUES ('{0}','{1}','{2}','{3}') ", procID, descrip, dura, cost);
            DataHandler dh = new DataHandler();
            dh.Insert(query);
        }

        public void EditProcedure(int procID, string descrip, int dura, double cost)
        {
            string query = string.format("UPDATE MedicalProcedure SET Description = '{1}',Duration = '{2}',Cost = '{3}' WHERE ProcedureID = {0}", procID, descrip, dura, cost);
            DataHandler dh = new DataHandler();
            dh.Update(query);
        }

        public void RemoveProcedure(int procID)
        {
            string query = string.format("DELETE FROM MedicalProcedure WHERE ProcedureID = {0}", procID);
            DataHandler dh = new DataHandler();
            dh.Delete(query);
        }

        public void ScheduleProcedure(DateTime dateOfPro, int animalId, int empId)
        {
            string query = string.format("INSERT INTO MedicalProcedures (EmployeeID, AnimalID, Date) VALUES ('{0}','{1}','{2}')", empID, animalId, dateOfPro);
            DataHandler dh = new DataHandler();
            dh.Insert(query);
        }

        public void ListProcedureByDate(DateTime dateOfPro)
        {
            string query = string.format("SELECT * FROM MedicalProcedures WHERE(Date: {2})",dateOfPro);
            DataHandler dh = new DataHandler();
            dh.Select(query);
        }

        #endregion






    }




}
